types/react-virtualized-select/index.d.ts

```diff
@@ -26,16 +26,17 @@ export interface VirtualizedOptionRenderOptions<T> {
  * Dummy interface to allow `VirtualizedSelectProps` to have an `optionRenderer` type
  * incompatible with the one in `ReactSelectProps`.
  */
-interface VirtualizedSelectPropsBase extends ReactSelectProps {
+interface VirtualizedSelectPropsBase<TValue = OptionValues> extends ReactSelectProps<TValue> {
     optionRenderer?: any;
 }
-export interface VirtualizedSelectProps extends VirtualizedSelectPropsBase {
+export interface VirtualizedSelectProps<TValue = OptionValues> extends VirtualizedSelectPropsBase<TValue> {
     async?: boolean;
+    loadOptions?: LoadOptionsHandler<TValue>;
```

Comment:
This prop already exists in `ReactSelectProps` so you should be able to remove it.

Reply:
AFAIK, it only exists in ReactAsyncSelectProps not in ReactSelectProps. Since the latter is used as a base class, it won't allow using loadOptions.

Reply:
Ooh, I see. So there's an issue here that the value of the `async` property determines the required type of the other props.

I think this can be captured sort of like this:

```ts
interface AdditionalVirtualizedSelectProps {
  maxHeight?: number;
  ... etc ...
}

type VirtualizedSelectProps<TValue = ...> = ({
  async?: false | undefined;
} & ReactSelectProps<TValue>) | ({
  async: true;
} & ReactAsyncSelectProps<TValue>);
```

This _should_  get the job done, but I haven't actually tried it out. Not sure of the specifics since there's that comment on `VirtualizedSelectPropsBase` talking about overrides. Since you've added this option, would you mind trying this out and seeing if it works for you?

Reply:
That didn't do the trick. I've now added explicit async and creatable versions just like react-select did.